core/src/main/java/org/elasticsearch/index/mapper/CompletionFieldMapper.java

```diff
@@ -560,7 +560,7 @@ private void parse(ParseContext parseContext, Token token, XContentParser parser
                 }
             }
         } else {
-            throw new ElasticsearchParseException("failed to parse expected text or object got" + token.name());
+            throw new ElasticsearchParseException("failed to parse [" + parser.currentName() + "][" + parser.getTokenLocation() + "]: expected text or object, but got " + token.name());
```

Comment:
I wonder if it would make sense to use ParsingException otherwise, which supports token location natively.

Reply:
Oh yes, that's nicer than putting it in the string. Thanks.